### PR TITLE
Physical backup support proposal

### PR DESCRIPTION
**Pull request by sinbadonline**
[K8SPSMDB-713](https://jira.percona.com/browse/K8SPSMDB-713) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

MongoDB Operator implementation proposal to support physical backup function (K8SPSMDB-713)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| We run 0 out of 30|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/983/commits/60edac9098aae4e25109b12a9f3164cc5738cb8e
image: `perconalab/percona-server-mongodb-operator:PR-983-60edac90`

File: docs/architecture/decisions/0003-support-physical-backup.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## Context
+
+PBM tool can support physical backup from v1.7.0 while current Mongo operator can not. Considering the efficiency of physical backup for large size db, should enable this feature via Mongo operator.
+
+
+## Brief procedure
+
+* Add a type field (logical or physical) in CR PSMDB (spec.backup.type) and PSMDB-Backup to indicate backup type.
+* Let pbm tool recognize and check Mongo server's encryption/compression setting so that it can handle db data correctly during restoration. If pbm found different setting of encryption/compression, it will failed with error. PBM tool can not transfer db data files from one format to another for now.
+* Let pbm tool hide Mongo server's K8S entrypoint file (/data/db/ps-entry.sh) during restoration and recover entrypoint file before restoration succesful return. In this way, mongod container will stay in unready state and will not be restarted to jump in and interrupt restoration process.
```
Comment (hors) [inline docs/architecture/decisions/0003-support-physical-backup.md:20]: We have physical backup/restore in our PXC operator maybe we can use some logic from it. 
Maybe it is better not to ‘hide’ any entrypoints but add the logic to this entrypoint. E.g. entrypoint can understand that this is physical recovery case and do not start mongod. Or we can use the same logic as in PXC operator. We will have a job which will start specific pod with own entrypoint and perform the restore on PVC. @egegunes what do you think?

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| We run 0 out of 30|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/983/commits/60edac9098aae4e25109b12a9f3164cc5738cb8e
image: `perconalab/percona-server-mongodb-operator:PR-983-60edac90`

**Comment by egegunes** [comment]
Hi @sinbadonline, physical backups and restores are merged to main in https://github.com/percona/percona-server-mongodb-operator/pull/1112. Thank you for your proposal, I'm closing this one.

**Pull request closed without merging**